src/Symfony/Component/Intl/Data/Generator/ZoneDataGenerator.php

```diff
@@ -0,0 +1,172 @@
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    protected function generateDataForLocale(BundleReaderInterface $reader, $tempDir, $displayLocale)
+    {
+        $localeBundle = $reader->read($tempDir, $displayLocale);
+
+        if (isset($localeBundle['zoneStrings']) && null !== $localeBundle['zoneStrings']) {
+            $data = array(
+                'Version' => $localeBundle['Version'],
+                'Names' => self::generateZones(
+                    $reader->read($tempDir, 'timezoneTypes'),
+                    $reader->read($tempDir, 'metaZones'),
+                    $reader->read($tempDir, 'root'),
+                    $localeBundle
+                ),
+            );
+
+            $this->zoneCodes = array_merge($this->zoneCodes, array_keys($data['Names']));
+
+            return $data;
+        }
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    protected function generateDataForRoot(BundleReaderInterface $reader, $tempDir)
+    {
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    protected function generateDataForMeta(BundleReaderInterface $reader, $tempDir)
+    {
+        $rootBundle = $reader->read($tempDir, 'root');
+
+        $this->zoneCodes = array_unique($this->zoneCodes);
+
+        sort($this->zoneCodes);
+
+        $data = array(
+            'Version' => $rootBundle['Version'],
+            'Zones' => $this->zoneCodes,
+        );
+
+        return $data;
+    }
+
+    private static function generateZones(ArrayAccessibleResourceBundle $typeBundle, ArrayAccessibleResourceBundle $metaBundle, ArrayAccessibleResourceBundle $rootBundle, ArrayAccessibleResourceBundle $localeBundle): array
+    {
+        $available = array();
+        foreach ($typeBundle['typeMap']['timezone'] as $zone => $_) {
+            if (self::UNKNOWN_ZONE === $zone) {
+                continue;
+            }
+
+            $available[$zone] = true;
+        }
+
+        $metazones = array();
+        foreach ($metaBundle['metazoneInfo'] as $zone => $info) {
+            foreach ($info as $metazone) {
+                $metazones[$zone] = $metazone->get(0);
+            }
+        }
+
+        $zones = array();
+        foreach (array_keys($available) as $zone) {
+            $name = null;
+            $city = null;
+
+            if (isset($localeBundle['zoneStrings'][$zone])) {
+                $data = $localeBundle['zoneStrings'][$zone];
+                $name = $data->get('ls') ?? $data->get('lg') ?? $data->get('ld');
+                $city = $data->get('ec');
+            } elseif (isset($rootBundle['zoneStrings'][$zone])) {
+                $data = $rootBundle['zoneStrings'][$zone];
+                $name = $data->get('ls') ?? $data->get('lg') ?? $data->get('ld');
+                $city = $data->get('ec');
+            }
+
+            if (null === $city && 0 !== strrpos($zone, 'Etc:') && false !== $i = strrpos($zone, ':')) {
+                $city = str_replace('_', ' ', substr($zone, $i + 1));
```

Comment:
i need to check if this fallback is still needed. Ive added the root bundle fallback above later, and AFAIK it makes this one redundant.

status: needs work

Reply:
for future referece, this solves

```diff
-        "Africa\/Abidjan": "Greenwich Mean Time (Abidjan)",
-        "Africa\/Accra": "Greenwich Mean Time (Accra)",
-        "Africa\/Addis_Ababa": "East Africa Time (Addis Ababa)",
-        "Africa\/Algiers": "Central European Standard Time (Algiers)",
+        "Africa\/Abidjan": "Greenwich Mean Time",
+        "Africa\/Accra": "Greenwich Mean Time",
+        "Africa\/Addis_Ababa": "East Africa Time",
+        "Africa\/Algiers": "Central European Standard Time",
```

i think it's more explicit to always include the example city (`ec`), and i believe for these cases when it's not provided by intl we can infer it from here.